core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigInstantiator.java

```diff
@@ -60,7 +61,16 @@ public static void handleObject(Object o) {
         }
 
         final Class<?> cls = o.getClass();
-        final String name = dashify(cls.getSimpleName().substring(0, cls.getSimpleName().length() - clsNameSuffix.length()));
+        final String name;
+        ConfigRoot configRoot = cls.getAnnotation(ConfigRoot.class);
+        if (configRoot != null && !configRoot.name().equals(ConfigItem.HYPHENATED_ELEMENT_NAME)) {
+            name = configRoot.name();
+            if (name.startsWith("<<")) {
+                throw new IllegalArgumentException("Found unsupported @ConfigRoot.name = " + name + " on " + cls);
```

Comment:
I had a quick look at other default names and didn't see how to support them properly. So better complain loudly here.

Reply:
We have a few that are using `@ConfigRoot.name = ConfigItem.PARENT`, e.g. `CommandLineRuntimeConfig`.
But those are very special and it doesn't make sense to pass them to `ConfigInstantiator`, AFAICS.